### PR TITLE
Show fewer colors on main page, aggregate by agency

### PR DESCRIPTION
**Pull request by zachmargolis**
Based on some Slack feedback from @nickttng that there are a lot of colors by default

This turns the "unfiltered" report into one that aggregates by agency, so that selecting agency is more of a drilldown, so it changes the table as well

But now that I look at this, I think I want to keep the aggregation in the table and maybe leave out the color changing

| mode | before | after |
| --- | --- | -- |
| no agency selected | Screen Shot 2021-09-14 at 3 17 01 PM |  Screen Shot 2021-09-14 at 3 16 49 PM |
| single agency selected [no change] | Screen Shot 2021-09-14 at 3 17 23 PM | Screen Shot 2021-09-14 at 3 17 35 PM |

### CODE REVIEW

File: src/daily-auths-report.tsx [hunk 1]
```diff
@@ -120,6 +120,38 @@ function tabulate(
...
+
+  const header = ["Agency", "IAL", ...days.map(yearMonthDayFormat), "Total"];
+
+  const body = Array.from(rolledup)
+    .sort(([agencyA], [agencyB]) => ascending(agencyA, agencyB))
+    .flatMap(([agency, ials]) =>
+      Array.from(ials).map(([ial, ialDays]) => {
+        const dayCounts = days.map((date) => ialDays.get(yearMonthDayFormat(date)) || 0);
+
+        return [agency, String(ial), ...dayCounts, dayCounts.reduce((d, total) => d + total, 0)];
```
Comment (aduth) [inline src/daily-auths-report.tsx:146]: Why does only the `ial` number need to be string-ified, and not the subsequent cells? Or does it?

File: src/daily-auths-report.tsx [hunk 2]
```diff
@@ -173,7 +205,7 @@ function DailyAuthsReport(): VNode {
                   value: "date",
                   thresholds: utcDay,
                 },
-                fill: agency ? "friendly_name" : "agency",
+                fill: agency ? "friendly_name" : "steelblue",
                 title: (bin: ProcessedResult[]) => {
```
Comment (aduth) [inline src/daily-auths-report.tsx:209]: Should we be returning a title anymore for the "All" agencies?

File: src/daily-auths-report.tsx [hunk 3]
```diff
@@ -190,10 +222,13 @@ function DailyAuthsReport(): VNode {
   return (
     <div>
       <div className="chart-wrapper" ref={ref} />
-      <Table data={tabulate(data || [], agency, ial)} numberFormatter={format(",")} />
+      <Table
+        data={agency ? tabulate(data || [], agency, ial) : tabulateSumByAgency(data || [], ial)}
```
Comment (aduth) [inline src/daily-auths-report.tsx:226]: Maybe `results` should be an optional argument?

```suggestion
        data={agency ? tabulate(data, agency, ial) : tabulateSumByAgency(data, ial)}
```

**Review by aduth — APPROVED** [review]

File: src/daily-auths-report.tsx (hunk 1 above)
Comment (zachmargolis) [reply src/daily-auths-report.tsx:146]: Numbers get right aligned and formatted with commas, and I figured IAL didn't need that treatment

**Review by nickttng — APPROVED** [review]
I hadn't anticipated my comment to be taken as a feedback. 😂

The after version of "No agency selected" makes sense to me while keeping the "single agency selected" as is

File: src/daily-auths-report.tsx (hunk 2 above)
Comment (zachmargolis) [reply src/daily-auths-report.tsx:209]: Probably not! good catch

**Comment by zachmargolis** [comment]
Related, I'd love to make the table linkable, ex clicking on an agency in the table updates the form to select that specific agency, unsure of a good way to do that... will come back to the idea in a future PR

File: src/daily-auths-report.tsx (hunk 2 above)
Comment (zachmargolis) [reply src/daily-auths-report.tsx]: Updated in d72a3828a891f5d514e7ac1036678ccf1a8ab01c

File: src/daily-auths-report.tsx (hunk 3 above)
Comment (zachmargolis) [reply src/daily-auths-report.tsx]: Good idea, updated in 81de3fa60db6c6c20c85a9d5303c4b991c838a55

**Comment by zachmargolis** [comment]
Ok so I have changed my mind... I think it's nice to have a visual breakdown of the agencies on the first page, but still keep to fewer colors... so maybe more like this, gonna give this a shot, it's essentially two different pages now

IMG_0017

**Comment by zachmargolis** [comment]
Added in faceting by agency when all agencies are showing

Screen Shot 2021-09-15 at 10 23 26 AM

**Pull request merged**